service/src/main/java/uk/gov/hmcts/reform/fpl/service/DischargeCareOrderService.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.Party;
+import uk.gov.hmcts.reform.fpl.model.order.generated.GeneratedOrder;
+import uk.gov.hmcts.reform.fpl.model.order.selector.Selector;
+
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.IntStream.range;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DischargeCareOrderService {
+
+    public List<GeneratedOrder> getCareOrders(CaseData caseData) {
+        return caseData.getOrderCollection().stream()
+            .map(Element::getValue)
+            .filter(go -> ("Final care order".equals(go.getType()) || "Interim care order".equals(go.getType())))
```

Comment:
as mentioned somewhere else - as we have nice key like values for types, why we couldn't use them somehow to filtering?

Reply:
the problem is that GeneratedOrder does not have these keys, and there is no key for final care order, there is separate key for order type and separate key for subtype